src/ServiceControl.Plugin.Nsb5.Heartbeat/Heartbeats.cs

```diff
@@ -49,7 +49,11 @@ public void Dispose()
         {
             if (heartbeatTimer != null)
             {
-                heartbeatTimer.Dispose();
+                using (var manualResetEvent = new ManualResetEvent(false))
+                {
+                    heartbeatTimer.Dispose(manualResetEvent);
+                    manualResetEvent.WaitOne();
```

Comment:
i think that is reasonable given the context of the action it is trying to perform